### ISSUE TITLE
Thank you for this

### ISSUE THREAD

**Issue by bamboowonder** [EirikWulff/php-timelapse-creator#1]
Hey, just wanted to say thank you for this.  It matched exactly what I needed for a sky camera project I'm creating.  Without having to depend on some external overpriced website.
I was just thinking recently to ask about updating some of the dependencies and I see you just did.
cheers,
David

**Comment by TimeLaps3** [comment]
Hi there, did it work, could you share the url of the project? thanks a lot

**Comment by EirikWulff** [comment]
@bamboowonder, thanks for your feedback. Saw it ages ago, plainly forgot to reply, though, sorry about that. 

The code is not used actively in our setup anymore, as the building project we filmed finished some time ago. It still works, even on PHP 8.x. 

Feel free to run 'composer update' on your copy of the code, it should work just fine.

**Comment by bamboowonder** [comment]
after I wrote the comment, i came to the realization it wouldn't actually work for the solution I needed.  I needed to keep this very lean and simple with minimal external dependencies.   We are very constrained by power and bandwidth, camera accessibility.  so each photo needs to be carefully sized, compressed and timed.    

we ended up using a google pixel, with macrodroid to run the automation,  OpenCamera to take the photos, SyncthingFork to transfer the photos back to the server via a cellular 3g connection.

Then in the interest of KISS, (keep it simple stupid), on the server:
I used _h5ai to create a very pretty and simple directory listing for the photos
and every 15min I'd run the following to create a timelapse video with the photos.

`ffmpeg -pattern_type glob -i 2021-03-06_12\*.jpg -c:v libx264 -crf 28 -preset slow -profile:v high -movflags +faststart -vf zoompan=d=1:fps=4,framerate=4:interp_start=100:interp_end=150:scene=50 timelapsefilename.mp4`

And use this to create a timestamp watermark:

`mogrify -font Liberation-Sans -fill white -undercolor '#00000080' -pointsize 24 -gravity SouthEast -annotate +10+10 %t timelapsefilename.mp4`

in the end there are subdirectories for month/date and nearly live photos with periodically generated timelapses.

It worked well for our needs.   The site is private though so I can't share a link here.

If there was going to be a next optimization,it would be to process the timelapse video on the phone, as a video of a 1000 images is a significantly smaller filesize than the sum of 1000 photos.   So we could achieve higher quality and/or use less cellular data and get more frequent updates.   Initial testing showed trying to get ffmpeg running with the parameters I needed, on the phone, was a pain.  It also used a lot more power, which is an issue as the whole thing is run on a small solar panel.  And the project is contrained by cost and time, both of which are mine.
